index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
> You think it's worth attaching keys and tokens now?

If you check if out, you'll see that `parse` returns an array of keys and strings, then the `toRegexp` functionality turns that into a regexp and just an array of keys. Since there's actually no need for the strings at that moment. I do think it's worth returning both in the parse step since there's additional logic that implementors can easily get wrong otherwise (for example, un-escaping characters). I think it's valuable to provide the most functionality we can right out of the box, since it doesn't add much overhead here (we're already matching for escaped characters).